src/Microsoft.AspNetCore.ResponseCaching/ResponseCachingContext.cs

```diff
@@ -227,9 +249,25 @@ internal void FinalizeCachingBody()
         {
             if (State.ShouldCacheResponse && ResponseCacheStream.BufferingEnabled)
             {
-                State.CachedResponse.Body = ResponseCacheStream.BufferedStream.ToArray();
+                if (ResponseCacheStream.BufferedStream.Length >= _options.MinimumSplitBodySize)
```

Comment:
Offline discussion: Move the sharding to an implementation detail of the IResponseCache. E.g. there's no reason to shard for memory cache.